airflow/executors/kubernetes_executor_utils.py

```diff
@@ -129,7 +129,7 @@ def _run(
     ) -> str | None:
         self.log.info("Event: and now my watch begins starting at resource_version: %s", resource_version)
 
-        kwargs = {"label_selector": f"airflow-worker={scheduler_job_id}"}
+        kwargs = {"label_selector": "airflow-worker"}
```

Comment:
By modifying the label selector to `airflow-worker`, the executor will watch all pods created by Airflow. This change introduces a slight additional load, but it is essential to address a critical bug in the executor that currently disrupts its high-availability mode. IMO, this adjustment should not pose any issues.

However, it is crucial to include a comment preceding this statement clarifying the reason behind watching all pods. This comment will prevent anyone from reverting the change under the false assumption that it improves performance.